CustomPlugin/CpController.cs

```diff
@@ -1,19 +1,44 @@
-﻿using CustomPlugin.Environment;
+using Decorators;
+using Environment;
 using Microsoft.AspNetCore.Mvc;
 
 namespace CustomPlugin
 {
     [ApiController]
     [Route("[controller]")]
-    public class CpController : ControllerBase
+    public abstract class CpController : ControllerBase
     {
         protected readonly MiaEnvConfiguration MiaEnvConfiguration;
         protected readonly ServiceClientFactory ServiceClientFactory;
+        protected readonly DecoratorResponseFactory DecoratorResponseFactory;
 
-        public CpController(MiaEnvConfiguration miaEnvConfiguration, ServiceClientFactory serviceClientFactory)
+        public CpController(MiaEnvConfiguration miaEnvConfiguration, ServiceClientFactory serviceClientFactory,
+            DecoratorResponseFactory decoratorResponseFactory)
         {
             MiaEnvConfiguration = miaEnvConfiguration;
             ServiceClientFactory = serviceClientFactory;
+            DecoratorResponseFactory = decoratorResponseFactory;
+        }
+        
+        [HttpGet]
+        [Route("/-/healthz")]
```

Comment:
I worry that this might cause the same problem happened in [springboot library](https://github.com/mia-platform/custom-plugin-java-springboot/issues/5).

Should we apply the same strategy?

Reply:
Solved with same strategy.